src/charm.py

```diff
@@ -38,8 +39,8 @@ def _on_demo_server_pebble_ready(self, event):
         self.unit.status = ops.ActiveStatus()
 
     @property
-    def _pebble_layer(self):
-        """Return a dictionary representing a Pebble layer."""
+    def _pebble_layer(self) -> ops.pebble.Layer:
+        """Return a Layer object representing a Pebble layer."""
```

Comment:
This was a terrible docstring before (apart from it being outdated). Maybe we could take this opportunity to improve it? Something like:

```suggestion
        """A Pebble layer for the FastAPI demo services."""
```